tests/Utilities/MakeWithRandomValues.hpp

```diff
@@ -167,3 +156,24 @@ ReturnType make_with_random_values(
   fill_with_random_values(make_not_null(&result), generator, distribution);
   return result;
 }
+
+/// \ingroup TestingFrameworkGroup
+/// \brief Make a fixed-size data structure and fill with random values
+///
+/// \details Given a template argument type `T`, create an object of the same
+/// type, fills it with random values, and returns the result. Acts as a
+/// convenience function to avoid users needing to put in constructors with
+/// `signaling_NaN()`s themselves when making with random values.
+///
+/// used as
+/// `make_with_random_values<Type>(make_not_null(&gen),make_not_null(&dist)`
+template <typename T, typename UniformRandomBitGenerator,
+          typename RandomNumberDistribution>
+T make_with_random_values(
+    const gsl::not_null<UniformRandomBitGenerator*> generator,
+    const gsl::not_null<RandomNumberDistribution*> distribution) noexcept {
+  auto result = static_cast<T>(
```

Comment:
Why do you need to do a cast here? I think what you want is for integral types to use their `max()` value instead, but I'm not sure.

Reply:
The cast is to optionally call the constructor of a fixed-size container type which can be created with a single value of the contained type, or just use the type itself if it is fundamental. If the cast is insufficiently safe, I can make a checker for `TypeTraits.hpp` to verify that the object can be so constructed and `static_assert` before this line

Reply:
Then I think you should do a safe conversion/construction using `T result{...};` (braces are important)